### ISSUE TITLE
Generated tsconfig.json file should not list files.

### ISSUE THREAD

**Issue by dbaeumer** [angular/angular-cli#684]
1. Windows 10 
2. Versions. Please run `ng --version`. 
```
angular-cli: 1.0.0-beta.1
node: 4.2.6
os: win32 x64
```

Steps
- create a new project using ng new project
- switch to the directory and open VSCode 1.0.0 on it
- open `project`.component.spec.ts and then `project`.component.ts

Observe that you get errors about unsupported decorators in the second file. The reason is that the generated tsconfig.json file has a files sections only including main.ts and typings.d.ts. Either the files section must list all src files or better be removed to cover all *.ts files automatically. Otherwise the files under src will not honor the tsconfig.json settings.

**Comment by bentolor** [comment]
Generated output causes issues in IntelliJ IDEA as well: 
- With the generated `files` section it warns that it won't compile .ts files, as they are not listed in `tsconfig.json`
- Without the `files` section IDEA is happy, but basic tasks like `ng server` or `ng test` fail with cryptic stacktraces containing a _"Cannot find name 'module'."_ as hint.

**Comment by rkralston** [comment]
I have created ts configuration files that work in VS Code and Broccoli. I posted them here: https://github.com/Microsoft/vscode/issues/8954#issuecomment-231960894.

It appears the two sets of settings are mutually exclusive. If I apply the VS Code Settings, Broccoli breaks. If I apply the Broccoli Settings, VS Code throws errors.

I can post more of my project if that will help. I can also give an individual access to our BitBucket repo.

**Comment by rkralston** [comment]
I see a solution here for IntelliJ IDEA. 

I'm not sure how I'm supposed to configure VS Code so the Editor and Broccoli are both happy.

Maybe I'm too thick to see the solution?

I tried looking at Broccoli for help. I have no idea where to start. They have hundreds of projects. Sorry to be such a newb.

**Comment by rkralston** [comment]
Based on my conversations with Microsoft, the files attribute is a violation of the current standard for TS. This is the same issue that IDEA is having. It will also not work out of the box. 

How would I raise this as an issue with Broccoli? 

Does Broccoli have a newer compiler to which we can upgrade which is standards compliant?

Exactly which TS compiler is being used?

**Comment by rkralston** [comment]
I have looked at broccoli-typescript.js and it looks like you are using that setting to filter the *.d.ts files that are used by the incremental compile. I commented out lines 59-61. That made me clean up the typings directory. However, I still have two problems: 

1. I am using the jQuery typings definitions which appear to conflict with protractor's definition of '$' which is a little weird. 
2. I am getting an error on the definition of modules asking me to use NodeModule. I am guessing this one of the reasons you wanted to filter the *.d.ts files?
